pkg/defaults/init.go

```diff
@@ -38,7 +61,8 @@ func InitIntegration(portClient *cli.PortClient, applicationConfig *port.Config)
 		}
 
 		klog.Warningf("Could not get integration with state key %s, error: %s", applicationConfig.StateKey, err.Error())
-		if err := integration.CreateIntegration(portClient, applicationConfig.StateKey, applicationConfig.EventListenerType, defaultIntegrationConfig); err != nil {
+		_, err := integration.CreateIntegration(portClient, applicationConfig.StateKey, applicationConfig.EventListenerType, defaultIntegrationConfig, applicationConfig.CreatePortResourcesOrigin == port.CreatePortResourcesOriginPort)
```

Comment:
you could use a helper variable here instead of evaluating twice
```
shouldCreateResourcesUsingPort = applicationConfig.CreatePortResourcesOrigin == port.CreatePortResourcesOriginPort
```